### PR TITLE
Apply more strict parsing of actions in bulk API

### PR DESCRIPTION
**Pull request by PeteGillinElastic**
Previously, the following classes of malformed input were deprecated but not rejected in the action lines of the a bulk request:

 - Missing closing brace;
 - Additional keys after the action (which were ignored);
 - Additional data after the closing brace (which was ignored).

They will now be considered errors and rejected.

The existing behaviour is preserved in v8 compatibility mode.

(N.B. The deprecation warnings were added in 8.1. The normal guidance to deprecate for a whole major version before removing does not apply here, since this was never a supported API feature. There is a risk to the lenient approach since it results in input being ignored, which is likely not the user's intention.)

### CODE REVIEW

**Comment by PeteGillinElastic** [comment]
The `8.17.0 / bwc-snapshot` failure in CI appears to be because it's broken in main.

**Review by dakrone — APPROVED** [review]
LGTM, I left one comment. Thanks Pete!

File: docs/changelog/115923.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type: breaking
+issues: [ ]
+breaking:
+  title: Apply more strict parsing of actions in bulk API
+  area: REST API
+  details: >-
+    Previously, the following classes of malformed input were deprecated but not rejected in the action lines of the a
+    bulk request: missing closing brace; additional keys after the action (which were ignored); additional data after
+    the closing brace (which was ignored). They will now be considered errors and rejected.
+  impact: Users must provide well-formed input when using the bulk API
```
[the remark this answers is not in the document]
Comment (PeteGillinElastic) [reply docs/changelog/115923.yaml:13]: Good call. I'm adding "They can request REST API compatibility with v8 to get the previous behaviour back as an interim measure."

**Comment by leemthompo** [comment]
@PeteGillinElastic is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

**Comment by PeteGillinElastic** [comment]
> @PeteGillinElastic is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

I think this depends on the policy for the serverless changelog. Previously, we issued a warning on this class of malformed input but ignored it; now, we reject it. It would only break serverless users if they'd been writing malformed input and ignoring the warnings.

Also, this change was made before serverless GA.

**Pull request merged**